src/main/java/phi/Deadline.java

```diff
@@ -0,0 +1,47 @@
+package phi;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents the Deadline task type, and contains a String with a user-specified date as the deadline of a task
+ */
+public class Deadline extends Task {
+    private String deadlineString;
+
+    /**
+     * Constructor for a new Deadline instance
+     *
+     * @param msg       Task message to be displayed
+     * @param isDone    Boolean determining if task is completed
+     * @param deadline  User-specified deadline of task
+     */
+    public Deadline(String msg, boolean isDone, String deadline) {
+        super(Type.D,isDone, msg);
+        this.deadlineString = deadline;
+        LocalDate deadlineDate;
+        DateTimeFormatter inputFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        DateTimeFormatter outputFormat = DateTimeFormatter.ofPattern("MMM dd yyyy");
+
+        try {
+            deadlineDate = LocalDate.parse(deadlineString, inputFormat);
+            this.deadlineString = deadlineDate.format(outputFormat);
+        } catch (DateTimeParseException e) {
+            //System.out.println("Can't find a proper date format, using deadline as a String");
+        }
+    }
+
+    @Override
+    public String toString() {
+        return super.toString() + String.format(" (by: %s)", deadlineString);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String outputFormat() {
```

Comment:
Great job on isolating writing of output to file in a separate function. Perhaps, the name for the function can be a bit more explanative, indicating that it is used to return output for file writing, not to be confused with output for the chatbot itself. 🤔